File: 5sem/JiPP/Lab01/Task_4/src/task3.c
#include <stdio.h>
#include "task3_wrapper.h"

int main (int argc, char **argv) {
    int n, num1 = 0, num2 = 0, num3 = 0;
    sollution s;

    bool loop = true;
    while(loop) {
        printf("\n\nChoose the operation: \n [0] factorial\n [1] factorial recur\n [2] gcd\n [3] gcd recur\n [4] equation\n [5] equation recur\n else - Quit\n\n");
        scanf("%d", &n);
        switch (n) {
            case 0:            
                printf("Enter the number: \n");
                scanf("%d", &num1);
                printf("The factorial of %d equals %d", num1, factorial(num1));
                break;
            case 1:
                printf("Enter the number: \n");
                scanf("%d", &num1);
                printf("The factorial of %d equals %d", num1, factorial_recur(num1));
                break;
            case 2:        
                printf("Enter two numbers: \n");
                scanf("%d %d", &num1, &num2);
                printf("The GCD(%d, %d) equals %d", num1, num2, gcd(num1, num2));
                break;
            case 3:
                printf("Enter two numbers: \n");
                scanf("%d %d", &num1, &num2);
                printf("The GCD(%d, %d) equals %d", num1, num2, gcd_recur(num1, num2));
                break;
            case 4:
                printf("Enter three numbers: \n");
                scanf("%d %d %d", &num1, &num2, &num3);
                s = equation(num1, num2, num3);
                if (s.IsCorrect) {
                    printf("Sollution of: %dx + %dy = %d", num1, num2, num3);
                    printf("\nis: x = %d, y = %d", s.X, s.Y);
                }
                else { printf("Sollution does not exist in Integers"); }
                break;
            case 5:
                printf("Enter three numbers: \n");
                scanf("%d %d %d", &num1, &num2, &num3);
                s = equation_recur(num1, num2, num3);
                if (s.IsCorrect) {
                    printf("Sollution of: %dx + %dy = %d", num1, num2, num3);
                    printf("\nis: x = %d, y = %d", s.X, s.Y);
                }
                else { printf("Sollution does not exist in Integers"); }
                break;
            default:
                loop = false;
                break;
        }
    }
    return 0;
}

File: 5sem/JiPP/Lab01/Task_3/task3.py
import lab02_funs as lab02


def main():
    n: int; n1: int; n2: int; n3: int
    x: int; y: int; correct: bool
    stay_in: bool = True
    action: int
    while stay_in:
        print(
            "\n\nChoose the operation: \n [0] factorial\n [1] factorial recur\n [2] gcd\n [3] gcd recur\n [4] equation\n [5] equation recur\n else - Quit\n\n")
        action = int(input("action: "))
        if action == 0:
            print("Enter number: ")
            n1 = int(input("n = "))
            n = lab02.my_factorial(n1)
            print(f"Factorial of {n1}, is: {n}")
        elif action == 1:
            print("Enter number: ")
            n1 = int(input("n = "))
            n = lab02.my_factorial_recur(n1)
            print(f"Factorial of {n1}, is: {n}")
        elif action == 2:
            print("Enter numbers (a, b): ")
            n1 = int(input("a = "))
            n2 = int(input("b = "))
            n = lab02.my_gcd(n1, n2)
            print(f"GCD({n1}, {n2}) is: {n}")
        elif action == 3:
            print("Enter numbers (a, b): ")
            n1 = int(input("a = "))
            n2 = int(input("b = "))
            n = lab02.my_gcd(n1, n2)
            print(f"GCD({n1}, {n2}) is: {n}")
        elif action == 4:
            print("Enter numbers (a, b, c): ")
            n1 = int(input("a = "))
            n2 = int(input("b = "))
            n3 = int(input("b = "))
            x, y, correct = lab02.my_equation(n1, n2, n3)
            if correct:
                print(f"Solution of equation {n1}x + {n2}y = {n3}, is: x = {x}, y = {y}")
            else:
                print("Solution does not exist in Integers")
        elif action == 5:
            print("Enter numbers (a, b, c): ")
            n1 = int(input("a = "))
            n2 = int(input("b = "))
            n3 = int(input("b = "))
            x, y, correct = lab02.my_equation_recur(n1, n2, n3)
            if correct:
                print(f"Solution of equation {n1}x + {n2}y = {n3}, is: x = {x}, y = {y}")
            else:
                print("Solution does not exist in Integers")
        else:
            stay_in = False


if __name__ == '__main__':
    main()